src/app/LayoutCommon.astro

```diff
@@ -8,9 +8,25 @@ const { class: clazz, metaData } = Astro.props;
 
 const initData: any = {};
 
-const envKeys = [...Object.keys(process.env), ...Object.keys(import.meta.env)];
-envKeys.forEach((item) => {
-	initData[item] = process.env[item] || import.meta.env[item];
+// Step 1: List of keys that should NOT be exposed to the frontend
+const excludeKeys = ['PUBLIC_CRYPTO_PRIVATE_KEY', 'PUBLIC_KEYCLOAK_MANAGEMENT_CLIENT_ID', 'PUBLIC_KEYCLOAK_MANAGEMENT_CLIENT_SECRET', 'PUBLIC_ALLOW_DOMAIN'];
```

Comment:
Can we take this array from common constant file?

Reply:
Yes, I have taken the above array from the common constants file. This approach ensures better maintainability, as any new keys can be easily added to the array in a single location